indico/modules/events/registration/controllers/management/util.py

```diff
@@ -0,0 +1,24 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+
+def remove_regform_item_gaps(form_items, from_position):
+    """Update positions when deleting/disabling an item in order to prevent gaps"""
+    for item in form_items:
+        if item.position > from_position and item.is_enabled:
```

Comment:
Untested, but wouldn't something like this be safer? Unlike the current code it would ensure consistent positions on all items.

```python
positions = itertools.count(1)
disabled_positions = itertools.count(1)
for item in sorted(form_items, key=attrgetter('position')):
    pos = next(positions if item.is_enabled and not item.is_deleted else disabled_positions)
    item.position = pos
```

We could even go one step further and do this on the whole regform:

```
section_positions = itertools.count(1)
disabled_section_positions = itertools.count(1000)
for section in sorted(regform.sections, key=attrgetter('position')):
    section_active = section.is_enabled and not section.is_deleted
    section.position = next(section_positions if section_active else disabled_section_positions)
    # ensure consistent field ordering
    positions = itertools.count(1)
    disabled_positions = itertools.count(1000)
    for child in section.children:
        child_active = child.is_enabled and not child.is_deleted
        child.position = next(positions if child_active else disabled_positions)
```